### DISCUSSION TITLE
Project governance

### DISCUSSION THREAD

**Discussion by MattF-NSIDC** [earthaccess-dev/earthaccess#321] [General]
How can we move towards sustainable community governance for this project? Ideally, this project should be able to move forward regardless of whether someone at NSIDC has time to review or merge PRs. How do we get there?

* Develop our own governance plan?
* Seek to be adopted by an existing organization (PyOpenSci, Scientific Python, ???) who can provide governance expertise?
* ???

We'd love to hear ideas!

**Comment by mrocklin** [comment]
I'm generally cynical on governance models, so take what I'm about to say with some salt.

Project governance has a lot more to do with writing down what's actually true on the ground than it has to do with changing things.  It's rare for projects to change how they operate by doing governance-y things, especially early on.  If it does change things, the most common result is to increase inertia (which can sometimes be a good thing!).

In your situation my recommendation would be to not care about formal governance just yet, and instead focus on attracting maintainers to whom you can grant commit rights.  Based on what I'm hearing "people who can review and take responsible ownership over parts of the project" seems to be your primary bottleneck.  I would focus on that.  My guess is that developing a governance plan or being adopted by some other project does not contribute to the solution of this problem.

**Reply by MattF-NSIDC** [reply]
Thanks for sharing your perspective on this, Matt! 

>my recommendation would be to not care about formal governance just yet, and instead focus on attracting maintainers to whom you can grant commit rights

Do you have thoughts on tactics to achieve this? This is an area where NSIDC needs to develop experience. Bringing on James as a maintainer was a great step in the right direction, and in his case we didn't have to do any "attracting," he was ready to go!

**Reply by mrocklin** [reply]
🤷 make a thing that people need?  Seems like you've done a good job there ✅ 

When good people show up give them commit rights earlier than you're comfortable with.  That usually has a good effect of making people feel ownership and motivation.  

How to make more people show up?  No idea.  Advertise the project more to users?  Lots of people who we talk to who should know about this project don't.  I'd start there.  Developers tend to exist in user communities.

But really who knows.  I don't think that anyone has a recipe for this.

**Reply by MattF-NSIDC** [reply]
FWIW, I'm talking about "governance" in the most general sense. How do we not just add maintainers but empower our maintainers to make decisions without feeling like NSIDC needs to be directly involved?

**Reply by MattF-NSIDC** [reply]
>When good people show up give them commit rights earlier than you're comfortable with. That usually has a good effect of making people feel ownership and motivation.

:100:

**Reply by jhkennedy** [reply]
@MattF-NSIDC I don't think there's anything right now that *requires* NSIDC to be in the loop... is there anything that does?

I and @betolink are both admins^ and so can add collaborators/maintainers update secrets, etc., and from there, there are ~9 people who can approve and merge pull requests. Expanding the admin team is likely a good move. 

I think NSIDC being a gate right now is mostly cultural, which we might be able to shift with some small changes like setting up a `CODEOWNERS` so that all the maintainers are automatically requested for review, flushing out some of the developer/contributing docs, etc

---

^I could see NSIDC, at some point in the future, locking down admin rights in the NSIDC org. The fix in that case would be to move EarthAccess into it's own org or into an aligned FOSS org.
